lib/DAC/DAC.cpp

```diff
@@ -4,137 +4,165 @@
 
-    printf("start dac task\n");
+  // printf("start dac task\n");
 
-    while(1) {
+  while (1) {
 
-        // read user input
-        fgets_stdio_blocking(input_str, n);
+    // // read user input
+    // fgets_stdio_blocking(input_str, n);
 
-        printf("fgets read: %s\n", input_str);
+    // printf("fgets read: %s\n", input_str);
 
-        sscanf(input_str, "%d", &input);
+    // sscanf(input_str, "%d", &input);
 
-        if(input >= 0 && input < 256){
-            val = input;
-            printf("set dac to %d\n", val);
-        } else {
-            printf("dac: input value out of bound: %d\n", input);
-        }
+    // if (input >= 0 && input < 256) {
+    //   val = input;
+    //   printf("set dac to %d\n", val);
+    // } else {
+    //   printf("dac: input value out of bound: %d\n", input);
+    // }
 
-        // set potentiometer value
-        printf("set potentiometer value=%d\n", val);
-        set_pot(val, POT_CHAN0);
+    // // set potentiometer value
+    // printf("set potentiometer value=%d\n", val);
+    // set_pot(val, POT_CHAN0);
 
-        // sleep for 10s
-        vTaskDelay(100 / portTICK_PERIOD_MS);
-    }
+    // sleep for 10s
+    vTaskDelay(100 / portTICK_PERIOD_MS);
+  }
 }
 
-
-void init_dac() {
-    // nothing to do, i2c bus is getting initialized externally
+uint8_t get_cmd(pot_chan channel) {
+  uint8_t command = BASE_ADDR_CMD;
+  switch (channel) {
+  case POT_CHAN0:
+    command |= 0x01;
+    break;
+  case POT_CHAN1:
+    command |= 0x10;
+    break;
+  case POT_CHAN_ALL:
+    command |= 0x11;
+    break;
+  default:
+    command |= 0x01;
+    break;
+  }
+  return command;
 }
 
-uint8_t get_cmd(pot_chan channel){
-    uint8_t command = BASE_ADDR_CMD;
-    switch (channel) {
-        case POT_CHAN0:
-            command |= 0x01;
-            break;
-        case POT_CHAN1:
-            command |= 0x10;
-            break;
-        case POT_CHAN_ALL:
-            command |= 0x11;
-            break;
-        default:
-            command |= 0x01;
-            break;
-    }
-    return command;
-}
+// void set_pot(uint8_t val, pot_chan channel) {
+
+//   // setup command
+//   uint8_t command = get_cmd(channel);
+
+//   printf("Write motor potentiometer %02x to %d\n", command, val);
 
+//   // CRITICAL SECTION I2C: start
+//   xSemaphoreTake(i2c_mutex, portMAX_DELAY);
 
-void set_pot(uint8_t val, pot_chan channel){
+//   Wire.beginTransmission(I2C_ADDRESS_DS1803);
+//   Wire.write(command);
+//   Wire.write(val); // first pot value
+//   if (channel == POT_CHAN_ALL) {
+//     Wire.write(val); // second pot value
+//   }
+//   Wire.endTransmission();
 
-    // setup command
-    uint8_t command = get_cmd(channel);
+//   xSemaphoreGive(i2c_mutex);
+//   // CRITICAL SECTION I2C: end
+// }
 
-    // CRITICAL SECTION I2C: start
-    xSemaphoreTake(i2c_mutex, portMAX_DELAY);
+void set_pot(uint8_t value) {
+  printf("Write motor potentiometer to %d...", value);
+  // CRITICAL SECTION SPI: start
```

Comment:
I wouldn't comment out the body of the functions. Either we do not need them anymore (->remove them completely) or we still keep them in (-> deactivate/comment out the calling location).

Reply:
cleanup done